test/utils/test_generate.py

```diff
@@ -105,11 +99,46 @@ def test_sample(self, generation_model, mocker):
             x, max_seq_len=latent_seq_len, use_cache=True, causal=True
         )
         assert isinstance(out, SampleOutput)
-        actual = out.samples
-        assert_expected(
-            actual.shape, torch.Size([1, 3, 4, 8, 8])
-        )  # (b, c, *input_shape)
-        assert_expected(actual.sum().item(), -41.6888, rtol=1e-4, atol=1e-5)
+        expected = out.tokens
+        actual = torch.tensor(
+            [
+                477,
+                124,
+                624,
+                773,
+                559,
+                139,
+                213,
+                118,
+                364,
+                1014,
+                730,
+                871,
+                445,
+                982,
+                362,
+                942,
+                32,
+                536,
+                506,
+                280,
+                598,
+                81,
+                225,
+                689,
+                20,
+                661,
+                281,
+                782,
+                904,
+                595,
+                614,
+                815,
+            ]
+        ).unsqueeze(
+            0
+        )  # (b, out_seq_len)
```

Comment:
@langong347 I see `expected` is generated by torch.multinomial. When adding new implementation for exponential (which will be called by multinomial),  even the new exponential implementation fits the exponential distribution but it is not the same as the old one, and the result of multinomial will also change. Since the results of multinomial changed `assert_expected(actual, expected)` will raise error.
May I know can we use a more flexible check for the actual token and expected token ? For example : compute multiple times and use the mean values to compare.